src/engine/TransitivePath.cpp

```diff
@@ -187,7 +191,15 @@ size_t TransitivePath::getSizeEstimate() {
 }
 
 // _____________________________________________________________________________
-size_t TransitivePath::getCostEstimate() { return getSizeEstimate(); }
+size_t TransitivePath::getCostEstimate() {
+  auto costEstimate = getSizeEstimate();
+  for (auto* ptr : getChildren()) {
```

Comment:
// Add the cost for computing the left and right side (in case they are not variables).